helm-chart/images/binderhub/binderhub_config.py

```diff
@@ -84,7 +84,8 @@ def get_value(key, default=None):
     c.BinderHub.build_namespace = os.environ['BUILD_NAMESPACE']
 
 if c.BinderHub.auth_enabled:
-    hub_url = urlparse(c.BinderHub.hub_url)
+    hub_url = urlparse(c.BinderHub.hub_url_public if 'hub_url_public' in c.BinderHub
+                       else c.BinderHub.hub_url)
```

Comment:
Perhaps we can make hub_url_public have a `@default` decorated function returning self.hub_url? I think this logic is too messy to use whenever we want to use hub_url_public, which happen to only be once at the moment but who knows.